### DISCUSSION TITLE
How to perform free energy minimization well?

### DISCUSSION THREAD

**Discussion by biglinn** [SSCHAcode/python-sscha#350] [Q&A]
Hi, @mesonepigreco:
I do SSCHA iteration with manually submission. But the result of minimization is not satisfactory and even strange. When I plot the info during the minimization with `sscha-plot-data.py`, the figure is confusing, as follows:
![image](https://github.com/user-attachments/assets/e8fd395f-5a9b-4939-9a16-4d5f2a3baf0f)

It's very different from Au's figure in the first tutorial, which is as follows:
![image](https://github.com/user-attachments/assets/2535f05c-118f-4fd7-a58e-f36b57ead565)

Is it the reason for few minimization steps? Or is there other parameters about free energy minimization?

**Comment by mesonepigreco** [comment]
Hi, what is free energy, and what is its gradient? (Plotted in the other figure of sscha-plot-data.py) 
If the gradient is small, the result converges close to the starting point. If the starting point was the harmonic dynamical matrix, it is the signature that your system is harmonic; thus, the SSCHA result coincides with the harmonic result.

**Reply by biglinn** [reply]
This is the first iteration:
![image](https://github.com/user-attachments/assets/0851f8b0-9047-4191-8e0c-5673fb8fc5c3)

This is the 5th iteration:
![image](https://github.com/user-attachments/assets/71e139e7-92ff-44ac-b9cc-7bafd1668776)

So, according to the gradient, does it mean that the result converges in the 5th iteration?
That's fantastic !  I used to always judge convergence by plotting phonon dispersion and observing change between two iterations.
